dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/QaServiceTest.kt

```diff
@@ -0,0 +1,47 @@
+package org.dataland.e2etests.tests
+
+import org.awaitility.Awaitility.await
+import org.dataland.datalandbackend.openApiClient.model.CompanyAssociatedDataEuTaxonomyDataForNonFinancials
+import org.dataland.datalandbackend.openApiClient.model.QAStatus as BackendQaStatus
+import org.dataland.datalandqaservice.openApiClient.model.QAStatus as QaServiceQaStatus
+import org.dataland.e2etests.auth.TechnicalUser
+import org.dataland.e2etests.utils.ApiAccessor
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.util.concurrent.TimeUnit
+
+class QaServiceTest {
+    private val apiAccessor = ApiAccessor()
+
+    private val testDataEuTaxonomyNonFinancials = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getTData(1).first()
+
+    private val testCompanyInformation = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getCompanyInformationWithoutIdentifiers(1).first()
+
+    @Test
+    fun `post dummy data, qa it and check the qa status changes`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
```

Comment:
When I looked into the jwtHelper, I found a method `getSingleValueFromJsonStringForKey()`.
In fact, I wrote that some time ago.
Now today in a call with Siegfried I noticed, that using json-libraries, it is a lot easier to extract a specific value from a json string.
Shouldn't we use that instead of my more or less "custom" json-field-parser method?

Reply:
Possibly working alternative pushed